articles/azure-app-configuration/enable-dynamic-configuration-dotnet-core-push-refresh.md

```diff
@@ -175,6 +175,50 @@ A random delay is added before the cached value is marked as dirty to reduce pot
 > [!NOTE]
 > To reduce the number of requests to App Configuration when using push refresh, it is important to call `SetCacheExpiration(TimeSpan cacheExpiration)` with an appropriate value of `cacheExpiration` parameter. This controls the cache expiration time for pull refresh and can be used as a safety net in case there is an issue with the Event subscription or the Service Bus subscription. The recommended value is `TimeSpan.FromDays(30)`.
 
+## Optional: Incorporate synchronization tokens for real-time consistency
```

Comment:
With the new API, the `SetDirty` should be replaced. There is no reason for anyone still using the `SetDirty` API. I don't expect this to be an optional choice. We should just update the existing document.

Reply:
I think we should still keep the SetDirty option as the new APIs are only available from 5.x version onwards. If someone is on an older version, they can still implement push refresh without sync tokens.  Maybe we can switch the order? Talk about the sync token APIs first, and then mention SetDirty option in a "Note" for anyone using old provider?

Reply:
I would agree if the difference between the old and new APIs are just different ways to achieve the same result, but we know the `SetDirty` API could be problematic. Imagine what we will do if a customer comes to us and complains their app didn't pick up a config change after responding to the push notification - we will suggest them to upgrade to 5.x. So let's not put something we don't recommend in the doc. Let's push customers to 5.x.